algviz/linkedlist/linkedlistvisualizer.py

```diff
@@ -1,152 +1,152 @@
-		originalCenter = [center[0],center[1]]#so the two aren't connected (Changing one wont change the other now)
-		for i in ll:#draw nodes
-			offset = self.draw_contents(i, self.svg_doc, center, nested = True, getSize = True)#to get size, this will need to be replaced later with a function to just receive size w/o drawing
-			offset -= radius #only expand the bubble if its contents are greater than radius in size
-			if offset < 0:
-				offset = 0
-			offset_array.append(offset)
-			if nested:#note: this needs to be implemented better, i think we need more arguments to do this right, but for now i didn't want to overload on them
-				center[0] += offset/4
-				if not getSize:
-					self.svg_doc.add(self.svg_doc.ellipse(center = center, r = [radius/2 + offset/4,  radius/2], stroke_width = '3', stroke = 'black', fill = 'white'))
-					self.draw_contents(i, self.svg_doc, center, nested = True)
-				center[0] += 1.25*radius + offset/4#x coord of center moves right 100 + offset//2 px each iteration, we added offset earlier
-				size += 1.25*radius + offset/2
-			else:
-				center[0] += offset/2
-				if not getSize:
-					self.svg_doc.add(self.svg_doc.ellipse(center = center, r = [radius + offset/2, radius], stroke_width = '3', stroke = 'black', fill = 'white'))
-					self.draw_contents(i, self.svg_doc, center, nested = True)
-				center[0] += 2.5 *radius + offset/2#x coord of center moves right 100 + offset//2 px each iteration, we added offset earlier
-				size += 2.5*radius + offset
-		center = originalCenter#this is whatever the original center for the first node is
-		for i in range(0, len(offset_array)):#add edges
-			if i != length - 1:#dont add an edge on the last one
-				if nested:
-					center[0] += offset_array[i] / 4
-					if not getSize:
-						self.svg_doc.add(self.svg_doc.line(start = [center[0] + radius/2 + offset_array[i]/4, center[1]], end = [center[0] + (3 * radius) / 4 + offset_array[i]/4, center[1]], stroke_width = '3', stroke = 'black'))
-					center[0] += radius*1.25 + offset_array[i]/4
-				else:
-					center[0] += offset_array[i]/2
-					if not getSize:
-						self.svg_doc.add(self.svg_doc.line(start = [center[0] + radius + offset_array[i]/2, center[1]], end = [center[0] + (3 * radius) / 2 + offset_array[i]/2, center[1]], stroke_width = '3', stroke = 'black'))
-					center[0] += 2.5*radius + offset_array[i]/2
-		self.save()
-		return size
-
-	def __str__(self):
-		return self.svg_doc.tostring()
-
-class LLVisualiser(BaseVisualizer):
-	def __init__(self, svg_doc = None):
-		self.svg_doc = svg_doc
-
-	def _draw(self, ll, center = None, nested = False, getSize = False):
-		self.draw_contents(item = ll)
-
-if __name__ == "__main__":
-	ll = LinkedList()
-	ll2 = LinkedList()
-	n7 = Node(4, None)
-	n6 = Node(111, n7)
-	n8 = Node(0, n6)
-	ll2.addNode(n8)
-	n5 = Node(321, None)
-	n4 = Node(144123, n5)
-	n3 = Node(ll2, n4)
-	n2 = Node(2232, n3)
-	n1 = Node(123, n2)
-	ll.addNode(n1)
-
-
-	t = LLVisualiser()
-	t._draw(ll)
+import svgwrite
```

Comment:
I don't really want to include this code in the main part of our module because it doesn't use the APIs that we're planning to build.  (See the design document for our picture-drawing API in picture.md, in our shared architecture directory.)
On the other hand, maybe it belongs in the `algviz/tools` directory that I created in #9 .